### ISSUE TITLE
Bug with empty strings with $in clause

### ISSUE THREAD

**Issue by pierreis** [masumsoft/express-cassandra#76]
I keep getting errors when performing updates with a WHERE clause containing null string.

Here is an example, largely based on the home page one:

```javascript
let Cassandra = require('express-cassandra');
let models = Cassandra.createClient({
  clientOptions: {
    contactPoints: ['127.0.0.1'],
    protocolOptions: { port: 9042 },
    keyspace: 'mykeyspace',
    queryOptions: {consistency: Cassandra.consistencies.one}
  },
  ormOptions: {
    defaultReplicationStrategy : {
      class: 'SimpleStrategy',
      replication_factor: 1
    },
    migration: 'safe',
    createKeyspace: true
  }
});

models.connect(function (err) {
  if (err) throw err;

  let MyModel = models.loadSchema('Person', {
    fields:{
      name    : "text",
      surname : "text",
      age     : "int"
    },
    key:["name", "surname"]
  }, function(err, UserModel){
    models.instance.Person.update({name: "Doe", surname: {$in: ["John", "Jane", ""]}}, {age:15}, function (err) {
      console.log(err);
    })
  });
});
```

The error I get is:

```
{ apollo.model.update.dberror: Error during update query on DB -> TypeError: Not a valid text value, expected String obtained { query_segment: '?', parameter: '' }
  at f (/mnt/hgfs/Workspace/Projects/foo/node_modules/express-cassandra/lib/orm/apollo_error.js:141:15)
  at /mnt/hgfs/Workspace/Projects/foo/node_modules/express-cassandra/lib/orm/base_model.js:1743:20
  at /mnt/hgfs/Workspace/Projects/foo/node_modules/express-cassandra/lib/orm/base_model.js:1239:9
  at next (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/utils.js:613:14)
  at RequestHandler.handleError (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/request-handler.js:316:12)
  at readCallback (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/request-handler.js:179:19)
  at Object.writeCallback [as callback] (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/connection.js:425:14)
  at whileProcess (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/writers.js:214:21)
  at next (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/utils.js:759:7)
  at Object.whilst (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/utils.js:749:3)
  at WriteQueue.process (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/writers.js:196:9)
  at WriteQueue.run (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/writers.js:190:10)
  at WriteQueue.push (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/writers.js:185:8)
  at Connection.sendStream (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/connection.js:411:19)
  at RequestHandler.sendOnConnection (/mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/request-handler.js:176:19)
  at /mnt/hgfs/Workspace/Projects/foo/node_modules/cassandra-driver/lib/request-handler.js:164:10 name: 'apollo.model.update.dberror' }
```

With debug logging enabled, here is the update command generated:

```
  express-cassandra executing query: UPDATE "Person" SET "age"=? WHERE "name" = ? AND "surname" IN ?; with params: [15,"Doe",["John","Jane",{"query_segment":"?","parameter":""}]] +14ms
```

This is obviously wrong. The parameter for `surname` should be `["John","Jane",""]` instead of `["John","Jane",{"query_segment":"?","parameter":""}]`.

I would suspect a bug with a `null` test, returning `true` for empty strings. If I have time, I might try to look a bit into the root cause of the issue.

**Comment by pierreis** [comment]
Okay. It did bother me a bit not to understand.
I did pinpoint the location of the problem on [lines 1054-1055](https://github.com/masumsoft/express-cassandra/blob/master/src/orm/base_model.js#L1054-L1055) of `lib/orm/base_model.js`.

```javascript
if (_.isPlainObject(dbVal) && dbVal.parameter) return dbVal.parameter;
return dbVal;
```

If the parameter is the empty string, the test fails, causing `dbVal` to be returned.

**Comment by masumsoft** [comment]
@pierreis great catch! fixed now in version 1.6.3
